types/node/index.d.ts

```diff
@@ -2017,7 +2004,7 @@ declare module "url" {
         hash?: string;
     }
 
-    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
+    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): URL;
```

Comment:
Unfortunately not. They are implemented different. URL is a real ES6 class with getters/setters,... whereas Url is a function with direct properties (no prototype properties).
They look similar but there are differences.